lib/dfe/analytics/event.rb

```diff
@@ -78,6 +79,12 @@ def with_data(hash)
         self
       end
 
+      def with_hidden_data(hash)
+        @event_hash.deep_merge!(hidden_data: hash_to_kv_pairs(hash))
```

Comment:
If we have the `data` and `hidden_data` in in the original hash then we would need just the with_data methods as follows:

``` Ruby
def with_data(hash)
  @event_hash.deep_merge!(data: hash_to_kv_pairs(hash[:data]))
  @event_hash.deep_merge!(hidden_data: hash_to_kv_pairs(hash[:hidden_data]))
  self
end
```

Reply:
Now updated